sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConnectionScope.cs

```diff
@@ -886,17 +890,65 @@ public void Dispose()
 
             closeHandler = (snd, args) =>
             {
-                ActiveLinks.TryRemove(link, out var timer);
-
-                timer?.Change(Timeout.InfiniteTimeSpan, Timeout.InfiniteTimeSpan);
-                timer?.Dispose();
-
+                StopTrackingLinkAsActive(link);
                 link.Closed -= closeHandler;
             };
 
             link.Closed += closeHandler;
         }
 
+        /// <summary>
+        ///   Performs the actions needed to stop tracking the specified AMQP
+        ///   link as an active link bound to this scope.
+        /// </summary>
+        ///
+        /// <param name="link">The link to stop tracking.</param>
+        /// <param name="authorizationRefreshTimer">The timer used to manage refreshing authorization, if the link requires it.</param>
+        ///
+        /// <remarks>
+        ///   This method operates on the specified <paramref name="link"/> in order to remove it
+        ///   from active tracking; no assumptions are made about the open/connected state of the link nor are
+        ///   its communication properties modified.
+        /// </remarks>
+        ///
+        protected virtual void StopTrackingLinkAsActive(AmqpObject link,
+                                                        Timer authorizationRefreshTimer = null)
+        {
+            var activeTimer = default(Timer);
+
+            if (link != null)
+            {
+                ActiveLinks.TryRemove(link, out activeTimer);
+
+                if (activeTimer != null)
+                {
+                    try
+                    {
+                        activeTimer?.Change(Timeout.InfiniteTimeSpan, Timeout.InfiniteTimeSpan);
+                        activeTimer?.Dispose();
```

Comment:
Isn't activeTimer guaranteed to be non-null here?

Reply:
Yes;  yes, it is.  That's a remnant of an earlier form .  I'll get that fixed up.